patches/server/0926-More-Teleport-API.patch

```diff
@@ -5,10 +5,36 @@ Subject: [PATCH] More Teleport API
 
 
 diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
-index 9d89da66c0c5ab07252430eaff6d20e5e83db658..935c482be4bf3767e197a576d0e38eac51115dd4 100644
+index 9d89da66c0c5ab07252430eaff6d20e5e83db658..c869b6948db38b8e823a22b5a82d239b1fe90361 100644
 --- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
 +++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
-@@ -1660,11 +1660,17 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
+@@ -719,6 +719,12 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
+                         // there to avoid any 'Moved wrongly' or 'Moved too quickly' errors.
+                         // We only do this if the Event was not cancelled.
+                         if (!oldTo.equals(event.getTo()) && !event.isCancelled()) {
++                            // Paper start
++                            if (oldTo.getX() == event.getTo().getX() && oldTo.getY() == event.getTo().getY() && oldTo.getZ() == event.getTo().getZ()) {
++                                this.player.getBukkitEntity().setRotation(event.getTo().getYaw(), event.getTo().getPitch());
++                                return;
++                            }
++                            // Paper end
+                             this.player.getBukkitEntity().teleport(event.getTo(), PlayerTeleportEvent.TeleportCause.PLUGIN);
+                             return;
+                         }
+@@ -1544,6 +1550,12 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
+                                         // there to avoid any 'Moved wrongly' or 'Moved too quickly' errors.
+                                         // We only do this if the Event was not cancelled.
+                                         if (!oldTo.equals(event.getTo()) && !event.isCancelled()) {
++                                            // Paper start
++                                            if (oldTo.getX() == event.getTo().getX() && oldTo.getY() == event.getTo().getY() && oldTo.getZ() == event.getTo().getZ()) {
```

Comment:
See other comment about changing worlds

You can use `PlayerMoveEvent#hasChangedBlock` as shorthand for this conditional too.

Reply:
hasChangedBlock only let me know if the from/to are different but i need check if the "to" originally send in event its the same than to set by user (world,x,y,z) then this short not works for this case.